File: algo_for_cpp/217.contains-duplicate.cpp
/*
 * @lc app=leetcode id=217 lang=cpp
 *
 * [217] Contains Duplicate
 */

// @lc code=start
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_map<int,int> cnt;
        for (int x : nums){
            cnt[x]++;
            if (cnt[x]>1)
                return true;
        }
        return false;
    }
};
// @lc code=end

File: algo_for_cpp/217.contains-duplicate.py
#
# @lc app=leetcode id=217 lang=python3
#
# [217] Contains Duplicate
#

# @lc code=start
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        cnt =dict()
        for n in nums:
            cnt[n]=cnt.get(n,0)+1
            if cnt[n]>1:
                return True
        return False
            # @lc code=end

